File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/guild/GuildMembershipMessage.scala





















// Generated on 05/03/2013 10:34:51
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GuildMembershipMessage extends GuildJoinedMessage {
  override val messageId = 5835

  protected trait Message extends GuildJoinedMessage.Message with Serializable {
    override val messageId = GuildMembershipMessage.messageId


    override def serialize(buf: Buffer) {
      super.serialize(buf)

    }

  }

  override def deserialize(buf: Buffer) = {
    val guildInfo = GuildInformations.deserialize(buf)
    val memberRights = buf.readUInt()
    val enabled = buf.readBoolean()

    create(guildInfo, memberRights, enabled)
  }

}

object GuildMembershipMessage extends GuildMembershipMessage {
  case class MessageImpl(guildInfo: GuildInformations.TypeImpl, memberRights: UInt, enabled: Boolean) extends Message
  type Target = MessageImpl

  override def create(guildInfo: GuildInformations.TypeImpl, memberRights: UInt, enabled: Boolean) = MessageImpl(guildInfo, memberRights, enabled)

  def apply(guildInfo: GuildInformations.TypeImpl, memberRights: UInt, enabled: Boolean) = create(guildInfo, memberRights, enabled)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GuildMembershipMessage.java


// Generated on 05/08/2013 19:37:51
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GuildMembershipMessage extends GuildJoinedMessage {
    public static final int MESSAGE_ID = 5835;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    
    public GuildMembershipMessage() { }
    
    public GuildMembershipMessage(GuildInformations guildInfo, long memberRights, boolean enabled) {
        super(guildInfo, memberRights, enabled);
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
    }
    
}
